feat: add metadata explanation in group module (#1216)
cosmos/sdk-tutorials: tutorials/understanding-group/index.md

Create the group:

> Note that, according to the [group documentation](https://docs.cosmos.network/v0.46/modules/group/06_metadata.html), the metadata of group must be off-chain. This means you must put the IPFS CID of the group metadata in the `metadata` field. For learning more about IPFS check out the [IPFS tutorial](https://tutorials.cosmos.network/tutorials/how-to-use-ipfs).

Reviewer: ```suggestion
Note that, according to the [group documentation](https://docs.cosmos.network/v0.46/modules/group/06_metadata.html), the metadata of a group must be off-chain. This means you must put the IPFS CID of the group metadata in the `metadata` field. To learn more about IPFS, check out the [IPFS tutorial](https://tutorials.cosmos.network/tutorials/how-to-use-ipfs).
```

> Here is how the metadata of Alice and Bob looks like:

Reviewer: ```suggestion
Here is what the metadata of Alice and Bob looks like:
```

**group_metadata.json**:

```json
{
  "name": "Football Association",
  "description": "Best football association",
  "groupWebsiteURL": "https://www.footbal.club",

>   "groupForumURL": ""

Reviewer: Is the JSON metadata format standardised?

Author: Yes, see https://docs.cosmos.network/v0.46/modules/group/06_metadata.html. Btw, I have updated in my cosmos-sdk PR the camelcase to understand (which is more standard)

}
```

> 

Reviewer: ```suggestion

Which gives the IPFS CID of `QmbZAqeGApZTfFkVEL3SL1NG5no37B6b6bbkw5g7z5Pzga`.

```

That would be great if you could pin that one on a remote service.

Author: Will do!

Author: Done https://ipfs.io/ipfs/QmXNvNnHrX7weSyDLBNEv6YxnmwEUncmvG1z8HTxXEBnW1

Create the group:

> Note that, according to the [group documentation](https://docs.cosmos.network/v0.46/modules/group/06_metadata.html), the metadata of a group must be off-chain. This means you must put the IPFS CID of the group metadata in the `metadata` field. To learn more about IPFS, check out the [IPFS tutorial](https://tutorials.cosmos.network/tutorials/how-to-use-ipfs).

Reviewer: ```suggestion
Note that, according to the [group documentation](https://docs.cosmos.network/v0.46/modules/group/06_metadata.html), the metadata of a group must be off-chain. This means that, should you choose IPFS as your storage, you must put the IPFS CID of the group metadata in the `metadata` field. To learn more about IPFS, check out the [IPFS tutorial](https://tutorials.cosmos.network/tutorials/how-to-use-ipfs).
```

$ simd query group group-members $GROUP_ID
```

> Nice! Your group has `QmbZAqeGApZTfFkVEL3SL1NG5no37B6b6bbkw5g7z5Pzga` as metadata (which you can recall with the `group-info` command), Alice as group admin, and Alice and Bob as group members.

Reviewer: ```suggestion
Nice! Your group has `QmbZAqeGApZTfFkVEL3SL1NG5no37B6b6bbkw5g7z5Pzga` as metadata (which you can recall with the `group-info` command), Alice as group admin, and Alice and Bob as group members.

<HighlightBox type="note">

The members of your group understand that this hash is for IPFS. Now, if you want to make it intelligible to a wider audience, you could replace that with:

* `ipfs://QmbZAqeGApZTfFkVEL3SL1NG5no37B6b6bbkw5g7z5Pzga`, which, although more understandable, is not a standard but ought to be.
* Or `https://ipfs.io/ipfs/QmbZAqeGApZTfFkVEL3SL1NG5no37B6b6bbkw5g7z5Pzga` which is a valid URL.

</HighlightBox>
```

Author: I can add, but not sure about that. It depends on the support of the clients. @hxrts it is not clear from the metadata spec what the recommendation is. Would you know?

Reviewer 2: ipfs:// is a pretty common convention and I think it make sense to recommend use

Author: Alright, added!

$ simd query group group-members $GROUP_ID
```

Nice! Your group has `ipfs://QmXNvNnHrX7weSyDLBNEv6YxnmwEUncmvG1z8HTxXEBnW1` as metadata (which you can recall with the `group-info` command), Alice as group admin, and Alice and Bob as group members.

<HighlightBox type="note">

> The members of your group understand that this hash is for IPFS.
> Without IPFS, the metadata is viewable at <https://ipfs.io/ipfs/QmXNvNnHrX7weSyDLBNEv6YxnmwEUncmvG1z8HTxXEBnW1>.

Reviewer: ```suggestion

The members of your group understand that this hash is for IPFS. Without IPFS, the metadata is viewable at <https://ipfs.io/ipfs/QmXNvNnHrX7weSyDLBNEv6YxnmwEUncmvG1z8HTxXEBnW1>.

```

Nice! Your group has `ipfs://QmXNvNnHrX7weSyDLBNEv6YxnmwEUncmvG1z8HTxXEBnW1` as metadata (which you can recall with the `group-info` command), Alice as group admin, and Alice and Bob as group members.

<HighlightBox type="note">

The members of your group understand that this hash is for IPFS. Without IPFS, the metadata is viewable at <https://ipfs.io/ipfs/QmXNvNnHrX7weSyDLBNEv6YxnmwEUncmvG1z8HTxXEBnW1>.

> </HighlightBox>

Reviewer: ```suggestion

</HighlightBox>
```

HighlightBox needs the blank line to compile.

Author: Thanks! I never render the website 😅

Have the group administrator create the group policy with metadata that identifies it as one with a quick turnaround:

```sh
$ simd tx group create-group-policy $ALICE $GROUP_ID "{\"name\":\"quick turnaround\",\"description\":\"\"}" policy.json
```

> Note that this time, the metadata is on-chain an escaped JSON string. This is as per the [group documentation](https://docs.cosmos.network/v0.46/modules/group/06_metadata.html).

Reviewer: My understanding is that the _escaping_ is for the command line only. After it has been saved, there are no more `\`.

Author: No, it will stay escaped, as the `Metadata` field is a string: https://github.com/cosmos/cosmos-sdk/blob/5deb137f7a83506381472928d898cd823735aade/x/group/tx.pb.go#L73
